mgmt/rest/v2/plugin.go

```diff
@@ -244,13 +280,40 @@ func (s *apiV2) loadPlugin(w http.ResponseWriter, r *http.Request, _ httprouter.
 			default:
 				ec = 500
 			}
+			cleanUpTempFiles(rp)
 			Write(ec, rb, w)
 			return
 		}
+		defer cleanUpTempFiles(rp)
 		Write(201, catalogedPluginBody(r.Host, pl), w)
 	}
 }
 
+func isTLSEnabled(cert, key string) bool {
+	if cert != "" && key != "" {
+		return true
+	}
+	return false
+}
+
+func createTempFile(b []byte, fn string) (string, error) {
+	file, err := ioutil.TempFile(os.TempDir(), fn)
+	if err != nil {
+		return "", err
+	}
+	_, err = file.Write(b)
+	if err != nil {
+		return "", err
+	}
+	return file.Name(), nil
+}
```

Comment:
Please, check if creating temp files is necessary. As far I know, checking if the path points to existing file is enough
Example (taken from API v1):
```
if _, err := os.Stat(keyPath); os.IsNotExist(err) {
	e := errors.New("Error: given key file is not available")
	rbody.Write(500, rbody.FromError(e), w)
	return
}
```